toolchain/check/type_completion.cpp

```diff
@@ -806,7 +806,51 @@ static auto RequireIdentifiedNamedConstraints(
   return true;
 }
 
-auto RequireIdentifiedFacetType(Context& context,
+// Makes a copy of a Specific from one generic to apply to another given
+// `generic_id`, with a given `Self` argument appended to the end.
+static auto MakeCopyOfSpecificAndAppendSelf(
+    Context& context, SemIR::LocId loc_id, SemIR::SpecificId specific_id,
+    SemIR::GenericId generic_id, SemIR::InstId self_id) -> SemIR::SpecificId {
+  auto source_specific_args_id =
+      context.specifics().GetArgsOrEmpty(specific_id);
+  auto source_specific_args =
+      context.inst_blocks().Get(source_specific_args_id);
+  llvm::SmallVector<SemIR::InstId> arg_ids;
+  arg_ids.reserve(source_specific_args.size() + 1);
+  // Start with the enclosing arguments from the source Specific.
+  llvm::append_range(arg_ids, source_specific_args);
+  // Add the new `Self` argument.
+  arg_ids.push_back(self_id);
+  return MakeSpecific(context, loc_id, generic_id, arg_ids);
+}
+
+static auto GetRequireImplsSpecificSelf(Context& context,
```

Comment:
```suggestion
// Returns the `Self` from a `require impls` specific.
static auto GetRequireImplsSpecificSelf(Context& context,
```

Minor, but the language keywords here are hard for me to parse ("require" as noun versus verb, whether "require impls" is a full term or two separate, etc) -- a comment would make this easy.

Reply:
Done as "Returns the `Self` from the specific of a `require` declaration."